app/controllers/revise_auth/password_controller.rb

```diff
@@ -1,24 +1,18 @@
 class ReviseAuth::PasswordController < ReviseAuthController
-  before_action :validate_current_password, only: [:update]
+  before_action :authenticate_user!
 
   def update
     if current_user.update(password_params)
-      flash[:notice] = I18n.t("revise_auth.password_changed")
+      redirect_to profile_path, notice: I18n.t("revise_auth.password_changed")
+    else
+      flash[:alert] = I18n.t("revise_auth.incorrect_password")
+      render "revise_auth/registrations/edit", status: :unprocessable_entity
     end
-
-    redirect_to profile_path
   end
 
   private
 
   def password_params
-    params.require(:user).permit(:password, :password_confirmation)
-  end
-
-  def validate_current_password
-    unless current_user.authenticate(params[:current_password])
-      flash[:alert] = I18n.t("revise_auth.incorrect_password")
-      render "revise_auth/registrations/edit", status: :unprocessable_entity
-    end
+    params.require(:user).permit(:password, :password_confirmation, :password_challenge)
```

Comment:
Pushing this up in the merge commit, but this is one gotcha to `password_challenge`. Have to make sure you call `with_defaults(password_challenge: "")` otherwise the user can delete the challenge from the form and update their password since the challenge would be `nil` when missing.

Reply:
Ahh good catch! Need me to submit a change or are you able to do this before merging?

Reply:
I got it and added a test. 👍